book/db/README.md

```diff
@@ -0,0 +1,332 @@
+
+- CanonicalHeaders
+- HeaderTD
+- HeaderNumbers
+- Headers
+- BlockBodies
+- BlockOmmers
+- NonCanonicalTransactions
+- Transactions
+- TxHashNumber
+- Receipts
+- Logs
+- PlainAccountState
+- PlainStorageState
+- Bytecodes
+- BlockTransitionIndex
+- TxTransitionIndex
+- AccountHistory
+- StorageHistory
+- AccountChangeSet
+- StorageChangeSet
+- TxSenders
+- Config
+- SyncStage
+
+<br>
+
+## Database
+
+Reth's database design revolves around it's main [Database trait](https://github.com/paradigmxyz/reth/blob/0d9b9a392d4196793736522f3fc2ac804991b45d/crates/interfaces/src/db/mod.rs#L33), which takes advantage of [generic associated types](https://blog.rust-lang.org/2022/10/28/gats-stabilization.html) and [a few design tricks](https://sabrinajewson.org/blog/the-better-alternative-to-lifetime-gats#the-better-gats) to implement the database's functionality across many types.  Let's take a quick look at the `Database` trait and how it works.
+
+[File: crates/storage/db/src/abstraction/database.rs](https://github.com/paradigmxyz/reth/blob/main/crates/storage/db/src/abstraction/database.rs#L19)
+```rust ignore
+/// Main Database trait that spawns transactions to be executed.
+pub trait Database: for<'a> DatabaseGAT<'a> {
+    /// Create read only transaction.
+    fn tx(&self) -> Result<<Self as DatabaseGAT<'_>>::TX, Error>;
+
+    /// Create read write transaction only possible if database is open with write access.
+    fn tx_mut(&self) -> Result<<Self as DatabaseGAT<'_>>::TXMut, Error>;
+
+    /// Takes a function and passes a read-only transaction into it, making sure it's closed in the
+    /// end of the execution.
+    fn view<T, F>(&self, f: F) -> Result<T, Error>
+    where
+        F: Fn(&<Self as DatabaseGAT<'_>>::TX) -> T,
+    {
+        let tx = self.tx()?;
+
+        let res = f(&tx);
+        tx.commit()?;
+
+        Ok(res)
+    }
+
+    /// Takes a function and passes a write-read transaction into it, making sure it's committed in
+    /// the end of the execution.
+    fn update<T, F>(&self, f: F) -> Result<T, Error>
+    where
+        F: Fn(&<Self as DatabaseGAT<'_>>::TXMut) -> T,
+    {
+        let tx = self.tx_mut()?;
+
+        let res = f(&tx);
+        tx.commit()?;
+
+        Ok(res)
+    }
+}
+```
+Any type that implements the `Database` trait can create a database transaction, as well as view or update existing transactions. We already saw that the `Transaction` struct from the `stages` crate implements the `Database` trait, allowing it to store new headers, bodies and senders in the database during the Reth pipeline loop. In the code snippet below, you can see the `Transaction::open()` method, which uses the `Database::tx_mut()` function to create a mutable transaction. 
```

Comment:
`Transaction` is generic over a type that implements the `Database` trait, right? Doesn't implement it itself IIUC

Sorry to nit-pick, but I think we should be precise about wording when it comes to Rust semantics

Reply:
Thanks for catching this, I am updating all the language throughout the chapter to reflect this.